Source/Scene/PointCloud.js

```diff
@@ -0,0 +1,1387 @@
+        this._pointsLength = 0;
+        this._geometryByteLength = 0;
+
+        this._vertexShaderLoaded = options.vertexShaderLoaded;
+        this._fragmentShaderLoaded = options.fragmentShaderLoaded;
+        this._uniformMapLoaded = options.uniformMapLoaded;
+        this._batchTableLoaded = options.batchTableLoaded;
+        this._pickIdLoaded = options.pickIdLoaded;
+        this._opaquePass = defaultValue(options.opaquePass, Pass.OPAQUE);
+        this._cull = defaultValue(options.cull, true);
+
+        this.style = undefined;
+        this._style = undefined;
+        this.styleDirty = false;
+
+        this.modelMatrix = Matrix4.clone(Matrix4.IDENTITY);
+        this._modelMatrix = Matrix4.clone(Matrix4.IDENTITY);
+
+        this.time = 0.0; // For styling
+        this.shadows = ShadowMode.ENABLED;
+        this._boundingSphere = undefined;
+
+        this.clippingPlanes = undefined;
+        this.isClipped = false;
+        this.clippingPlanesDirty = false;
+
+        this.attenuation = false;
+        this._attenuation = false;
+
+        // Options for geometric error based attenuation
+        this.geometricError = 0.0;
+        this.geometricErrorScale = 1.0;
+        this.maximumAttenuation = this._pointSize;
+
+        initialize(this, options);
+    }
+
+    defineProperties(PointCloud.prototype, {
+        pointsLength : {
+            get : function() {
+                return this._pointsLength;
+            }
+        },
+
+        geometryByteLength : {
+            get : function() {
+                return this._geometryByteLength;
+            }
+        },
+
+        ready : {
+            get : function() {
+                return this._ready;
+            }
+        },
+
+        readyPromise : {
+            get : function() {
+                return this._readyPromise.promise;
+            }
+        },
+
+        color : {
+            get : function() {
+                return Color.clone(this._highlightColor);
+            },
+            set : function(value) {
+                this._highlightColor = Color.clone(value, this._highlightColor);
+            }
+        },
+
+        boundingSphere : {
+            get : function() {
+                if (defined(this._drawCommand)) {
+                    return this._drawCommand.boundingVolume;
+                }
+            },
+            set : function(value) {
+                this._boundingSphere = BoundingSphere.clone(value);
+            }
+        }
+    });
+
+    var sizeOfUint32 = Uint32Array.BYTES_PER_ELEMENT;
+
+    function initialize(pointCloud, options) {
+        var arrayBuffer = options.arrayBuffer;
+        var byteOffset = defaultValue(options.byteOffset, 0);
+
+        var uint8Array = new Uint8Array(arrayBuffer);
+        var view = new DataView(arrayBuffer);
+        byteOffset += sizeOfUint32;  // Skip magic
+
+        var version = view.getUint32(byteOffset, true);
+        if (version !== 1) {
```

Comment:
Hm.. I'm tempted to skip for now because `PointCloud` and `TimeDynamicPointCloud` don't have a list of extensions used by the tile/tileset. It would have to guess where to look for unsupported extension, like in the feature table and batch table extensions object. But it feels a bit weird to do that at the tile level.

Extension checking probably belongs at the `Cesium3DTileset` level. Maybe worth adding to https://github.com/AnalyticalGraphicsInc/cesium/pull/6780?